pkg/pb/metainfo.proto

```diff
@@ -193,3 +213,321 @@ message ProjectInfoRequest {
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    int32 part_number = 2;
+    int32 index = 3;
+
+    bytes encrypted_key_nonce = 4;
+    bytes encrypted_key = 5;
+
+    bytes checksum_encrypted_data = 6;
+    bytes encrypted_inline_data = 7;
+}
+
+message SegmentMakeInlineResponse {}
+
+message SegmentBeginDeleteRequest {
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    int32 part_number = 2;
+    int32 index = 3;
+}
+
+message SegmentBeginDeleteResponse {
+    bytes segment_id = 1;
+    repeated AddressedOrderLimit addressed_limits = 2;
+    
+    // TODO: should we include here bool finished for inline segments, or should we use batching to combine SegmentBeginDeleteRequest/SegmentFinishDeleteResponse
+}
+
+message SegmentFinishDeleteRequest {
+    bytes segment_id = 1;
+
+    // TODO: check for uplink not sending order limits to storage nodes
+}
+
+message SegmentFinishDeleteResponse {}
+
+message SegmentListRequest {
+    bytes stream_id          = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    int32 cursor_part_number = 2;
+    int32 cursor_index       = 3;
+    int32 limit              = 4;
+    // TODO: is there a neater way to express cursor
+}
+
+message SegmentListResponse {
+    repeated SegmentListItem items = 1;
+    bool more = 2;
+}
+
+message SegmentListItem { // TODO: should we rename this to SegmentIndex and use it elsewhere?
+    int32 part_number = 1;
+    int32 index = 2;
+}
+
+message SegmentDownloadRequest2 {
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    int32 part_number = 2;
+    int32 index = 3;
+
+    SegmentListItem next = 4;
+}
+
+message SegmentDownloadResponse2 {
+    bytes                           segment_id = 1;
+    repeated AddressedOrderLimit    addressed_limits = 2;
+}
+
+message BatchRequest {
+    oneof Request {
+        BucketCreateRequest         bucket_create = 1;
+        BucketGetRequest            bucket_get = 2;
+        BucketDeleteRequest         bucket_delete = 3;
+        BucketListRequest           bucket_list = 4;
+        BucketSetAttributionRequest bucket_set_attribution = 5;
+
+        ObjectBeginRequest          object_begin = 6;
+        ObjectCommitRequest         object_commit = 7;
+        ObjectListRequest           object_list = 8;
+        ObjectBeginDeleteRequest    object_begin_delete = 9;
+        ObjectFinishDeleteRequest   object_finish_delete = 10;
+
+        SegmentBeginRequest      segment_begin = 11;
+        SegmentCommitRequest     segment_commit = 12;
+        SegmentMakeInlineRequest segment_make_inline = 13;
+
+        SegmentBeginDeleteRequest  segment_begin_delete = 14;
+        SegmentFinishDeleteRequest segment_finish_delete = 15;
+
+        SegmentListRequest     segment_list = 16;
+        SegmentDownloadRequest segment_download = 17;
+    }
+    //repeated Request requests = 1;
```

Comment:
why are these repeated fields commented out?

Reply:
I forgot to fix this. done